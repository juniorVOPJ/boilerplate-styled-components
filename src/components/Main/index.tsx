import * as S from './styles'

const Main = ({
    title = 'Boilerplate Styled Components',
    description = 'NextJS, TypeScript, Styled Components e Storybook'
}) => (
    <S.Wrapper>
        <S.Logo
            src="/images/tech_logos.png"
            alt="Imagens de logotipos das tecnologias utilizadas no projeto."
        />
        <S.Title>{title}</S.Title>
        <S.Description>{description}</S.Description>
        <S.Illustration
            src="/images/hero-illustration.svg"
            alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
        />
    </S.Wrapper>
)

export default Main
