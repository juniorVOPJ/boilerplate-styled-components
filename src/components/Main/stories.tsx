import { Meta, StoryObj } from '@storybook/react'

import Main from '.'

export default {
    title: 'Main',
    component: Main,
    parameters: {
        layout: 'fullscreen',
        controls: {
            expanded: true
        }
    }
} as Meta

export const Default: StoryObj = {
    argTypes: {
        title: {
            control: 'text'
        },
        description: {
            control: 'text'
        }
    }
}

export const Basic: StoryObj = {
    args: {
        title: 'Título Básico',
        description: 'Descrição Básica'
    }
}
