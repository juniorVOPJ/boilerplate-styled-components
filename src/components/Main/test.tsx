import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
    it('should render the heading', () => {
        // renderiza o component
        const { container } = render(<Main />)

        // busca o elemento e verifica a existÃªncia dele
        expect(
            screen.getByRole('heading', {
                name: /boilerplate styled components/i
            })
        ).toBeInTheDocument()

        // gerar snapshot
        expect(container.firstChild).toMatchSnapshot()
    })
})
