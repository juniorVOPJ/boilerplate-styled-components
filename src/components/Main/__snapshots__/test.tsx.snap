// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem, 100%);
}

<main
  class="c0"
>
  <img
    alt="Imagens de logotipos das tecnologias utilizadas no projeto."
    class="c1"
    src="/images/tech_logos.png"
  />
  <h1
    class="c2"
  >
    Boilerplate Styled Components
  </h1>
  <h2
    class="c3"
  >
    NextJS, TypeScript, Styled Components e Storybook
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
    class="c4"
    src="/images/hero-illustration.svg"
  />
</main>
`;
